### PR TITLE
Updates readme

### PR DESCRIPTION
**Pull request by gokulchandra**
Updates Readme with new features.

### CODE REVIEW

**Comment by rybalkinsd** [comment]
Agreed. This data is important. Will sync with one additional change in retry and then merge

**Comment by gokulchandra** [comment]
Do you think we should start maintaining a CHANGELOG as well?

**Comment by rybalkinsd** [comment]
@gokulchandra I’m really thinking about moving docs to separate page/wiki, ‘cause it’s getting complicated to navigate readme file.
At least we need a Table of contents and a way to collapse examples.
Didn’t research options yet

File: README.md [hunk 1]
```diff
@@ -216,11 +216,16 @@ val response = httpDelete { }
...
 val fileUri = this.javaClass.getResource("/cat.gif").toURI()
 
 val response = upload {
+    headers {
+        ...
+        cookies {...}
+    }
+    params {...}
     url("http://postman-echo.com/post")
     file(fileUri)
```
Comment (rybalkinsd) [inline README.md]: As upload is more concentrated on `content` and `destination` I would recommend to put `headers { }` and `params { }` section right after `file( ... )`

**Comment by gokulchandra** [comment]
With respect to maintaining docs what are your thoughts on using `[Github Pages]` (https://pages.github.com/)

**Comment by rybalkinsd** [comment]
@gokulchandra Introduced [kohttp.gitbook.io](https://kohttp.gitbook.io)

**Pull request merged**